docs/sources/configure/_index.md

```diff
@@ -880,6 +880,40 @@ volume_results_cache:
   # compression. Supported values are: 'snappy' and ''.
   # CLI flag: -frontend.volume-results-cache.compression
   [compression: <string> | default = ""]
+
+# Cache series query results.
+# CLI flag: -querier.cache-series-results
+[cache_series_results: <boolean> | default = false]
+
+# If series_results_cache is not configured and cache_series_results is true,
+# the config for the results cache is used.
+series_results_cache:
```

Comment:
agree to this. one downside is that we would lose granularity with stats and metric collection since they are tried to the cache instance. labels, series and results caching all would be clubbed together.

Reply:
How valuable are the metrics? Could we use a different label based on which "subcomponent" is observing metrics?

Reply:
Good point. Thought about this when introducing it. Now we have so much repeated configs for  normal results cache (metric range queries), index stats, volume, labels, series and for metric instant queries in future. All have exactly same configs.

One downside is it's hard to clean this up without breaking changes I think. Other option is to introduce shared config with "subcomponent" label but only for labels and series without breaking existing configs, then we loose the consistency with how we configure different results cache.

Reply:
I would be fine with leaving this as is for now but combining them all in v3.0 with breaking changes, thoughts?

Reply:
Good idea. Adding it to the epic 👍

Reply:
another thing that I am worried about is blowing up the size of stats proto. does encoding ignore fields that are not set? if that's the case we don't have to worry about this

only one of these would be set for a given request, I guess we could just collapse this into a single field with a new cache type field
https://github.com/grafana/loki/blob/40d64ef6e79c04177efd25e99180d98ece0d7d2b/pkg/logqlmodel/stats/stats.proto#L40-L60